nes-data-types/tests/UnitTests/API/SchemaTest.cpp

```diff
@@ -359,4 +359,396 @@ TEST_F(SchemaTest, withoutSourceQualifierTest)
+    }
+
+    /// Test 5: Empty schemas - should return false
+    {
+        const Schema expectedSchema;
+        const Schema actualSchema;
+
+        auto diff = SchemaDiff::of(expectedSchema, actualSchema);
+        EXPECT_FALSE(diff.isDifferent());
+    }
+
+    /// Test 6: One empty schema - should return true
+    {
+        Schema expectedSchema;
+        expectedSchema.addField("field1", DataType::Type::INT64);
+
+        const Schema actualSchema;
+
+        auto diff = SchemaDiff::of(expectedSchema, actualSchema);
+        EXPECT_TRUE(diff.isDifferent());
+    }
+
+    /// Test 7: Complex scenario with multiple differences - should return true
+    {
+        Schema expectedSchema;
+        expectedSchema.addField("field1", DataType::Type::INT64);
+        expectedSchema.addField("field2", DataType::Type::FLOAT32);
+        expectedSchema.addField("field3", DataType::Type::BOOLEAN);
+
+        Schema actualSchema;
+        actualSchema.addField("field1", DataType::Type::INT32); /// Type mismatch
+        actualSchema.addField("field2", DataType::Type::FLOAT32); /// Same
+        /// field3 is missing
+        actualSchema.addField("field4", DataType::Type::VARSIZED); /// Extra field
+
+        auto diff = SchemaDiff::of(expectedSchema, actualSchema);
+        EXPECT_TRUE(diff.isDifferent());
+    }
+}
+
+TEST_F(SchemaTest, diffSchemasTest)
+{
+    /// Test 1: Identical schemas - should return empty diff
+    {
+        Schema expectedSchema;
+        expectedSchema.addField("field1", DataType::Type::INT64);
+        expectedSchema.addField("field2", DataType::Type::FLOAT32);
+        expectedSchema.addField("field3", DataType::Type::BOOLEAN);
+
+        Schema actualSchema;
+        actualSchema.addField("field1", DataType::Type::INT64);
+        actualSchema.addField("field2", DataType::Type::FLOAT32);
+        actualSchema.addField("field3", DataType::Type::BOOLEAN);
+
+        auto diff = SchemaDiff::of(expectedSchema, actualSchema);
+
+        EXPECT_TRUE(diff.missingFields.empty());
+        EXPECT_TRUE(diff.additionalFields.empty());
+        EXPECT_TRUE(diff.schemaMismatches.empty());
+    }
+
+    /// Test 2: Schema with missing fields in actual - should show missing field and the mismatch for the second fieĺd
+    {
+        Schema expectedSchema;
+        expectedSchema.addField("field1", DataType::Type::INT64);
+        expectedSchema.addField("field2", DataType::Type::FLOAT32);
+        expectedSchema.addField("field3", DataType::Type::BOOLEAN);
+
+        Schema actualSchema;
+        actualSchema.addField("field1", DataType::Type::INT64);
+        /// field2 is missing
+        actualSchema.addField("field3", DataType::Type::BOOLEAN);
+
+        auto diff = SchemaDiff::of(expectedSchema, actualSchema);
+
+        EXPECT_EQ(diff.missingFields.size(), 1);
+        EXPECT_EQ(diff.missingFields[0].name, "field3");
+        EXPECT_EQ(diff.missingFields[0].dataType.type, DataType::Type::BOOLEAN);
```

Comment:
This is problematic; this should just report that there is a size mismatch. Reporting that Field3 is missing is just misleading.